### ISSUE TITLE
Working folder definition in ProjectParameters.JSON

### ISSUE THREAD

**Issue by adityaghantasala** [KratosMultiphysics/Kratos#3504]
Is it possible to define a working folder in the JSON file and the all the information like MDPA, StructuralMaterials.json (for Structural mechanics) etc are obtained from this folder and also the output .bin file is written to this folder ? 

Is this feature available ? Is there a plan to do something like this ?

**Comment by loumalouomega** [comment]
This is already done in the tests, look a the configuration parameters of the tests in the StructuralMechanicsApplication. You will see as the path is specified (individually for each file), we can add a common "working_folder" so it is necessary to define only once

**Comment by adityaghantasala** [comment]
as you said how it is done in StructuralMechanicsApplication is a many changes. Also it will not work for output. Let say I want the output also in the folder ... is it possible like that ?

**Comment by loumalouomega** [comment]
Fortunately this is relatively easy to implement. But let's wait to the @KratosMultiphysics/structural-mechanics or even to the @KratosMultiphysics/technical-committee if we put in the base solver

**Comment by adityaghantasala** [comment]
ok .. but it would be great to have such feature.

**Comment by philbucher** [comment]
> Fortunately this is relatively easy to implement. But let's wait to the @KratosMultiphysics/structural-mechanics or even to the @KratosMultiphysics/technical-committee if we put in the base solver

Is it?
We just discussed it but I think you would have to manually add the name of the folders to the output-file-names

Or what would be your solution?

**Comment by loumalouomega** [comment]
Well in my head is easy, I need to check it. But basically the first level of reading is in the base analysis stage. Defining there the working folder, and then allowing it to the solvers to use it

**Comment by armingeiser** [comment]
Changing the working directory in python (like in the tests) would be the easiest solution to do so i think. No need of manipulation of the paths.

**Comment by adityaghantasala** [comment]
@armingeiser  you mean in the JSON file as done in tests ?

**Comment by armingeiser** [comment]
The tests (at least the ones i am using) always change the working directory to the location of the python test file. 

```python
with controlledExecutionScope(os.path.dirname(os.path.realpath(__file__))):
    # run the test
```

From here relative paths are used to open the json and also inside the json.
It depends on where you wan't to have your json file, if thats also in the remote folder, you can change the working directory already in the MainKratos.py before importing the json.

**Comment by philbucher** [comment]
> Changing the working directory in python (like in the tests) would be the easiest solution to do so i think. No need of manipulation of the paths.

Good idea, I think this should be doable

**Comment by loumalouomega** [comment]
Is this solved?
